### ISSUE TITLE
Bug from 3.1.1 to 3.1.2 FindClusters (Leiden)

### ISSUE THREAD

**Issue by chlee-tabin** [satijalab/seurat#2438]
Hi, I just wanted to leave a note that when I updated Seurat from 3.1.1 to 3.1.2, Leiden algorithm completely fails to generate expected clusters (although Louvain succeeds). Bumping back to 3.1.1 resolved the issue.

**Comment by chlee-tabin** [comment]
A reproducible example:

```
library(tidyverse)
library(Seurat)
pbmc_small %>% FindNeighbors( dims = 1:10 ) %>% FindClusters( algorithm = 4 ) %>% DimPlot()
sessionInfo()
```

v3.1.1 does not generate any warnings and generates 2 clusters. In 3.1.2+, the same code produces the following warning:

```
56 singletons identified. 9 final clusters.
```

Although the `pbmc_small` succeeds in generating clusters, (a) this is not the same behavior than Seurat v3.1.1 (b) in my data (more bigger), it completely fails to generate seurat clusters, or even breaks down failing the code.

The relevant diff https://github.com/satijalab/seurat/commit/49a1be0427f2f26a531eb468ba93eeb18d8a2edb#diff-36f063d32c470f47250912efbcfff4f3 shows some changes in code, but I cannot pinpoint what makes this difference.

**Comment by chlee-tabin** [comment]
I may be wrong but I think with a method="matrix", previously, it was calling as(object, "matrix") as input to ```leiden``` whereas in the v3.1.2+ it is calling `graph_from_adjacency_matrix`...

**Comment by andrewwbutler** [comment]
Fixed in #2440
